src/main/java/hudson/remoting/URLish.java

```diff
@@ -16,20 +18,28 @@
 abstract class URLish {
     private URLish() {}
 
+    /**
+     * Converts URLish to the standard {@link URL} type.
+     * @return URL or {@code null} if the target destination is known to be non-existent.
```

Comment:
@jglick After some consideration I feel returning null is a valid behavior. E.g. this code uses abstract URLish, which may be both URL and File pointers. I could refactor all the code to do the same with exceptions, but I do not feel it improves the code much. WDYT?

```java
public URL findResource(String name) {
        // first attempt to load from locally fetched jars
        URL url = super.findResource(name);
        final Channel channel = channel();
        if(url!=null || channel == null || !channel.isRemoteClassLoadingAllowed())   return url;

        try {
            if(resourceMap.containsKey(name)) {
                URLish f = resourceMap.get(name);
                if(f==null) return null;    // no such resource
                URL u = f.toURL();
                if (u!=null)    return u;
            }

```